models/servicenow__user_aggregated.sql

```diff
@@ -35,11 +35,11 @@ aggregate_counts as (
     select 
         sys_user.sys_user_id,
         sys_user.source_relation,
-        count(distinct user_grmember.sys_user_group_id) as count_distinct_sys_user_group_ids,
-        count(distinct user_has_role.sys_user_role_id) as count_distinct_sys_user_role_ids,
-        count(distinct user_role.sys_user_role_name) as count_distinct_sys_user_role_names,
-        count(distinct user_role.includes_roles) as count_distinct_included_roles,
-        count(distinct user_group.sys_user_group_roles) as count_distinct_sys_user_group_roles
+        count(distinct case when user_grmember.sys_user_group_id is not null and user_grmember.sys_user_group_id != '' then user_grmember.sys_user_group_id else null end) as count_distinct_sys_user_group_ids,
+        count(distinct case when user_has_role.sys_user_role_id is not null and user_has_role.sys_user_role_id != '' then user_has_role.sys_user_role_id else null end) as count_distinct_sys_user_role_ids,
+        count(distinct case when user_role.sys_user_role_name is not null and user_role.sys_user_role_name != '' then user_role.sys_user_role_name else null end) as count_distinct_sys_user_role_names,
+        count(distinct case when user_role.includes_roles is not null and user_role.includes_roles != '' then user_role.includes_roles else null end) as count_distinct_included_roles,
+        count(distinct case when user_group.sys_user_group_roles is not null and user_group.sys_user_group_roles != '' then user_group.sys_user_group_roles else null end) as count_distinct_sys_user_group_roles
```

Comment:
These case when statements look like recent updates. Would you be able to explain these changes? Why are we also checking a few fields in other tables to see if they are empty strings? 

Also, should we document this change in the CHANGELOG since it could impact the results from previous versions?

Reply:
Also be aware that these new fields are also brought into the `user_enhanced` model because of the select * on this dependent model. So we will need to update the CHANGELOG documentation to reflect this change as well.

Reply:
Yep, so this change was because I realized the original counts were incorrect due to unintentionally counting blank values, that were not null. I've added an example at the bottom of this [hex notebook](on aggregate_counts.sys_user_id = user_aggregates.sys_user_id). 

Therefore this `case when` update ensures those blank values are not included in the counts!

I will add this to the CHANGELOG! I've made sure they're in the ymls.